### ISSUE TITLE
Integrate Predictive Coding training mode into trainDir and evolveDir

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#1556]
## Summary

Wire the Predictive Coding inference and learning modules into NEAT-AI's existing `trainDir()` and `evolveDir()` pipelines as an optional training mode. When `predictiveCoding.enabled = true`, creatures can be trained using PC's local learning rules instead of (or in addition to) standard backpropagation.

Part of #1549
Depends on: #1555 (PC local learning rules)

### 1. PC Training Mode in `trainDir()`

Create `src/predictiveCoding/PredictiveCodingTrainer.ts` that orchestrates the full PC training loop:

```
For each training iteration:
  For each mini-batch of samples:
    For each sample in batch:
      1. Clamp inputs → run PC inference (settling)
      2. Accumulate weight/bias gradients from prediction errors
    3. Apply averaged weight updates (Hebbian learning)
  4. Compute validation error
  5. Check early stopping (targetError)
```

### 2. Integration Point in CreatureTraining

Modify `src/creature/CreatureTraining.ts` to:
- Check if `predictiveCoding.enabled` in the training options
- If enabled, delegate to `PredictiveCodingTrainer` instead of standard `BackPropagation`
- If disabled, existing training path is completely unchanged
- Return the same `TrainResult` interface for compatibility

### 3. Integration with `evolveDir()`

In the evolution loop:
- PC training is used during the per-creature training phase (same place where `trainDir()` is called)
- Scoring, selection, breeding, and mutation remain unchanged
- PC is purely a training strategy — it does not affect the evolutionary operators

### 4. Hybrid Mode (Optional Enhancement)

Consider a mode where PC inference runs first to provide better gradient signals, followed by a standard backprop pass. This is supported by Millidge et al. (2022c) showing PC approximates backprop.

Flag: `predictiveCoding.hybrid: boolean` (default: false)

### 5. Logging & Observability

- Log PC-specific metrics: total energy per iteration, convergence rate, average inference steps
- Integrate with existing `Logger` utility
- Include PC metrics in `TrainResult` or a new `PredictiveCodingResult`

## Acceptance Criteria

- [ ] `PredictiveCodingTrainer.ts` orchestrates full PC training loop
- [ ] `trainDir()` delegates to PC trainer when `predictiveCoding.enabled = true`
- [ ] `evolveDir()` works with PC training mode
- [ ] Existing training behaviour completely unchanged when PC is disabled
- [ ] `TrainResult` or compatible result returned from PC training
- [ ] Integration tests:
  - XOR problem solved via PC training (classic NEAT benchmark)
  - Simple regression task trained with PC
  - evolveDir with PC enabled produces improving fitness over generations
  - Hybrid mode (if implemented) produces at least equivalent results to backprop alone
- [ ] All existing tests pass unchanged
- [ ] `./quality.sh` passes
- [ ] Australian English spelling throughout

## Coding Practices Reminder
- TDD: Write the XOR integration test first, then make it pass with PC training
- DRY: Reuse existing data loading, scoring, and result types from `trainDir()`
- KISS: Start with pure PC mode; hybrid mode can be a separate follow-up if complex
